packages/web/src/components/apps/use-package-json.tsx

```diff
@@ -24,41 +31,69 @@ type ProviderPropsType = {
 export function PackageJsonProvider({ channel, children }: ProviderPropsType) {
   const [status, setStatus] = useState<NpmInstallStatus>('idle');
   const [output, setOutput] = useState<Array<OutputType>>([]);
+  const [nodeModulesExists, setNodeModulesExists] = useState<boolean | null>(null);
 
   const { addError } = useLogs();
 
+  useEffect(() => {
+    channel.push('deps:status', {});
+  }, [channel]);
+
+  useEffect(() => {
+    const callback = (data: DepsStatusResponsePayloadType) => {
+      setNodeModulesExists(data.nodeModulesExists);
+    };
+    channel.on('deps:status:response', callback);
+
+    return () => {
+      channel.off('deps:status:response', callback);
+    };
+  }, [channel]);
+
   const npmInstall = useCallback(
-    (packages?: Array<string>) => {
+    async (packages?: Array<string>) => {
       // NOTE: caching of the log output is required here because socket events that call callback
       // functions in here hold on to old scope values
```

Comment:
Note that `npmInstall` now returns a promise, so `await npmInstall(); foo` can be used to run something after a `npm install` has finished!